docs/dev/code-snippets.md

```diff
@@ -0,0 +1,53 @@
+<!--
+{% comment %}
+Copyright 2018-2020 IBM Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+# Elyra Code Snippets - Experimental
+
+The ability to reuse pieces of code allows users to avoid doing repetitive work, 
+making the programming workflow much more simple and productive.
+Elyra supports custom code snippets that can be added to the file editor.
+
+![Code Snippet Sample](../source/images/code-snippet-expanded.png)
+
+## Code Snippet data
```

Comment:
Should we have a small paragraph here and do something more detailed down below similar to how we tell people to create runtime? (I am not sure what's best here... as this is becoming very large)